### ISSUE TITLE
Error reading XML File

### ISSUE THREAD

**Issue by jlcanibe** [EasyBatch/easybatch-framework#47]
Hi, I´ve just tried parsing the tweets.xml file , but i got this:

```
19/04/2015 07:21:13 PM org.easybatch.core.impl.Engine call
INFO: Initializing easy batch engine
19/04/2015 07:21:13 PM org.easybatch.core.impl.Engine call
INFO: Data source: XML stream: java.io.FileInputStream@bb6ab6
19/04/2015 07:21:13 PM org.easybatch.core.impl.Engine call
INFO: Strict mode: false
19/04/2015 07:21:13 PM org.easybatch.core.impl.Engine call
INFO: Total records = 3
19/04/2015 07:21:13 PM org.easybatch.core.impl.Engine call
INFO: easy batch engine is running...
19/04/2015 07:21:13 PM org.easybatch.xml.XmlRecordReader hasNextRecord
GRAVE: An exception occurred during checking the existence of next xml record
javax.xml.stream.XMLStreamException: ParseError at [row,col]:[1,39]
Message: Read error
	at com.sun.org.apache.xerces.internal.impl.XMLStreamReaderImpl.next(XMLStreamReaderImpl.java:592)
	at com.sun.xml.internal.stream.XMLEventReaderImpl.peek(XMLEventReaderImpl.java:275)
	at org.easybatch.xml.XmlRecordReader.nextTagIsRootElementStart(XmlRecordReader.java:155)
	at org.easybatch.xml.XmlRecordReader.hasNextRecord(XmlRecordReader.java:83)
	at org.easybatch.core.impl.Engine.call(Engine.java:123)
	at easybatch.jaxb.XmlJaxbParser.execute(XmlJaxbParser.java:29)
	at main.Test.main(Test.java:10)
19/04/2015 07:21:13 PM org.easybatch.core.impl.Engine call
INFO: Shutting down easy batch engine
```

The contents of the tweets.xml file:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<tweets>
    <tweet>
        <id>1</id>
        <user>foo</user>
        <message>Hello</message>
    </tweet>
    <tweet>
        <id>2</id>
        <user>bar</user>
        <message>Hi!</message>
    </tweet>
    <tweet>
        <id>3</id>
        <user>moo</user>
        <message>Hola</message>
    </tweet>
</tweets>
```

This is my test:

```java
package easybatch.jaxb;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import org.easybatch.core.api.Report;
import org.easybatch.core.impl.Engine;
import org.easybatch.core.impl.EngineBuilder;
import org.easybatch.xml.XmlRecordReader;

public class XmlJaxbParser {
	private String xmlTag;
	private String xmlFileName;
	
	public XmlJaxbParser(String tag, String fileName)  {
		xmlTag = tag;
		xmlFileName = fileName;
	}
	
	public void execute() {
		try {
			
			Engine engine = new EngineBuilder()
			.reader(new XmlRecordReader(xmlTag, new FileInputStream(xmlFileName)))
			.build();
			
	        // Run the batch engine
	        Report report = engine.call();

	        // Print the batch execution report
	        System.out.println(report);
	        
		} catch (FileNotFoundException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
	}
}
```

```java
package main;
import easybatch.jaxb.XmlJaxbParser;

public class Test {
	public static void main(String[] args) {
		XmlJaxbParser parser = new XmlJaxbParser("tweet", "tweets.xml");
		parser.execute();
	}
}
```

I've tried with different encodings, files, etc. But can´t make it work.

**Comment by benas** [comment]
Hi,

Thank you for pointing out this issue with all details. In fact, there is a problem with the XmlRecordReader.

I'll fix that and come back to you asap.

Regards,
Mahmoud

**Comment by benas** [comment]
Hi @jlcanibe ,

I have deployed version 3.0.1-SNAPSHOT in maven central repository with the fix for this issue.

You can find in [this test](https://github.com/EasyBatch/easybatch-framework/blob/master/easybatch-xml/src/test/java/org/easybatch/xml/XmlIntegrationTest.java) a complete example of xml processing.

Could you please rerun your example with this new version and keep me informed if it is ok for you?

Many thanks upfront.
Mahmoud

**Comment by jlcanibe** [comment]
Hi benas,

Can you help me with the maven url of the snapshot repository?

**Comment by benas** [comment]
Hi,

It is here:

[https://oss.sonatype.org/content/groups/public/org/easybatch/](https://oss.sonatype.org/content/groups/public/org/easybatch/)

If you add the snapshot dependency to your pom, it should work fine. If it doesn't, you can add the following repository:

```xml
<repositories>
        <repository>
            <id>ossrh</id>
            <url>https://oss.sonatype.org/content/repositories/snapshots</url>
        </repository>
    </repositories>
```

Hope it helps.

Mahmoud

**Comment by jlcanibe** [comment]
Sorry, I posted this in the Json Reader issue...

Hi Mahmoud,

Sorry for the delay, everything worked fine, this is my test´s output:

Batch Report:
Start time = 2015-04-24 07:45:16
End time = 2015-04-24 07:45:16
Status = FINISHED
Batch duration = 32ms
Data source = XML stream: java.io.FileInputStream@bb6ab6
Total records = 3
Filtered records = 0 (0.0%)
Ignored records = 0 (0.0%)
Rejected records = 0 (0.0%)
Error records = 0 (0.0%)
Success records = 3 (100.0%)
Record processing time average = 10.666667ms

Keep on with easybatch, really nice component, im trying easyrules right now.
Thank you

**Comment by benas** [comment]
Cool! I'm glad everything is ok now :-)

The same problem happened with the Json reader, so I created an issue and fixed it in the same time.

I'll publish a new release 3.0.1 very soon with all those fixes. Keep tuned.

Kind regards
Mahmoud

**Comment by benas** [comment]
Hello @jlcanibe,

FYI, this fix has been released in version 3.0.1.

Regards
Mahmoud
